### PR TITLE
Add Inference test for llama

### PR DESCRIPTION
**Pull request by yuehuayingxueluo**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [ ] I have performed a self-review of my code
- [ ] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: tests/test_infer/llama_infer_eigine.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+        block_loc = torch.empty(self.bs, self.input_len + self.output_len, dtype=torch.long, device="cuda")
+        start_loc = torch.zeros(self.bs, dtype=torch.int32, device="cuda")
+        seq_len = torch.zeros(self.bs, dtype=torch.int32, device="cuda")
+        max_total_token_num = self.bs * (self.input_len)
+        max_len_in_batch = self.input_len 
+        for i in range(self.bs):
+            block_loc[i, 0:self.input_len] = i * self.input_len + torch.arange(0, self.input_len, dtype=torch.int32, device="cuda")
+            start_loc[i] = i * self.input_len
+            seq_len[i] = self.input_len
+        print(block_loc.shape)
```
Comment (CjhHa1) [inline tests/test_infer/llama_infer_eigine.py:119]: del `print` here

File: tests/test_infer/llama_infer_eigine.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+        self.model = self.model.half()
+        self.model.to(torch.cuda.current_device())
+        
+
+    def build_model(self):
+        # create new model
+        self.orgin_model = self.model
+        shardconfig = ShardConfig(
+            tensor_parallel_process_group=self.pg_mesh.get_group_along_axis(2),
+            enable_tensor_parallelism=True,
```
Comment (CjhHa1) [inline tests/test_infer/llama_infer_eigine.py:144]: you can add `inference only` = true

File: tests/test_infer/llama_infer_eigine.py [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+        )
+        shardformer = ShardFormer(shard_config=shardconfig)
+        
+        if self.bs >= 4:
+            self.init_and_insert_cache_manager()
+            policy = LlamaModelInferPolicy()
+        else:
```
Comment (CjhHa1) [inline tests/test_infer/llama_infer_eigine.py:151]: we still need to use infer policy even if cache manager is not needed?

**Review by tiandiao123 — CHANGES_REQUESTED** [review]

**Review by tiandiao123 — APPROVED** [review]
LGTM！

**Pull request merged**